### PR TITLE
Add an integration test for null-only columns

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

A follow-up of https://github.com/apache/druid/pull/12279 to add integration tests for querying null-only columns using SQL.

<hr>

##### Key changed/added classes in this PR
 * `ITNilColumnTest`

<hr>

This PR has:
- [x] been self-reviewed.
- [x] added integration tests.

### CODE REVIEW

File: integration-tests/src/test/java/org/apache/druid/tests/indexer/ITNilColumnTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+{
+  private static final Logger LOG = new Logger(ITNilColumnTest.class);
+  private static final String NIL_DIM1 = "nilDim1";
+  private static final String NIL_DIM2 = "nilDim2";
+
+  private final List<String> dimensions;
+
+  public ITNilColumnTest()
+  {
+    this.dimensions = new ArrayList<>(DEFAULT_DIMENSIONS.size() + 2);
```
Comment (somu-imply) [inline integration-tests/src/test/java/org/apache/druid/tests/indexer/ITNilColumnTest.java:59]: Is there a particular reason we add nil dimensions only at start at end and not in the middle ?

**Review by somu-imply — APPROVED** [review]
LGTM. One minor comment

File: integration-tests/src/test/java/org/apache/druid/tests/indexer/ITNilColumnTest.java (hunk 1 above)
Comment (jihoonson) [reply integration-tests/src/test/java/org/apache/druid/tests/indexer/ITNilColumnTest.java:59]: Hmm, there was no particular reason, but that case is tested in unit tests such as https://github.com/apache/druid/blob/master/extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIndexTaskTest.java#L450. Dimension order doesn't do much in this test though since the columns in the `columns` table of the information schema is ordered lexicographically than the original order in the segment.

**Review by clintropolis — APPROVED** [review]
